goblet/deploy.py

```diff
@@ -265,3 +274,127 @@ def zip_directory(
         for path in globbed_files:
             if not set(path.parts).intersection(exclusion_set):
                 self.zipf.write(str(path))
+
+
+class RevisionSpec:
+    def __init__(self, config={}, versioned_clients=None, name="goblet"):
+        self.versioned_clients = versioned_clients
+        config = GConfig(config=config)
+        self.cloudrun_configs = config.cloudrun or {}
+        self.cloudrun_revision = config.cloudrun_revision or {}
+        self.req_body = {}
+        self.latestArtifact = ""
+        self.name = name
+
+    # calls latest build and checks for its artifact to avoid image:latest behavior with cloud run revisions
+    def getArtifact(self):
+        defaultProject = get_default_project()
+        buildClient = self.versioned_clients.cloudbuild
+        resp = buildClient.execute(
+            "list", parent_key="projectId", parent_schema=defaultProject, params={}
+        )
+        latestBuildId = resp["builds"][0]["id"]
+        resp = buildClient.execute(
+            "get",
+            parent_key="projectId",
+            parent_schema=defaultProject,
+            params={"id": latestBuildId},
+        )
+        self.latestArtifact = (
+            resp["results"]["images"][0]["name"]
+            + "@"
+            + resp["results"]["images"][0]["digest"]
+        )
+
+    # splits traffic proportionaly from already deployed traffic
+    def modifyTraffic(self):
```

Comment:
can we add a test for this function?
initilize RevisionSpec and then call modifyTraffic and we can test the resp. it might be better to seperate this out into two function one that gets the `initial service config` and then this function is purely logic on determining the traffic. that will make this test much easier since we wouldnt need to mock any api calls then.